avm/res/network/dnsresolver/inbound-endpoint/main.bicep

```diff
@@ -0,0 +1,56 @@
+metadata name = 'DNS Resolver Inbound Endpoint'
+metadata description = 'This module deploys a DNS Resolver Inbound Endpoint.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Private DNS Resolver.')
+@minLength(1)
+param dnsResolverName string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Required. The subnet ID of the inbound endpoint.')
+param subnetId string
+
+@description('Optional. The private IP address of the inbound endpoint.')
+param privateIpAddress string?
+
+@description('Optional. The private IP allocation method of the inbound endpoint.')
+param privateIpAllocationMethod string = 'Dynamic'
+
+@description('Required. The name of the inbound endpoint.')
+param name string
+
+resource dnsResolver 'Microsoft.Network/dnsResolvers@2022-07-01' existing = {
+  name: dnsResolverName
+}
+
+resource dnsResolver_inboundEndpoint 'Microsoft.Network/dnsResolvers/inboundEndpoints@2022-07-01' = {
+  parent: dnsResolver
+  name: name
+  location: location
+  tags: tags
+  properties: {
+    ipConfigurations: [
```

Comment:
I guess this is a 1:1 port from CARML right? If not, I'd be wondering if it's possible to only configure a single ip config per endpoint

Reply:
I recalled this not being possible beforehand - Having tested it, this is also correct, if you supply multiple IP's for the single endpoint this results in a exception being thrown from `Microsoft.Azure.Networking.Dns.ManagedResolver.Frontend.Contracts.Exceptions.Service.FrontendBadRequestLegacyServiceException`. The permitted experience is multiple InboundEndpoints with single IP Configurations. I believe the Array format comes as a requirement from frontDoor configuration for `Microsoft.Azure.Networking`